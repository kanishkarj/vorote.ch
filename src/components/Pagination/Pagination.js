// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import styled, { css } from 'styled-components';
import { PAGINATION } from '../../constants';
import { marginTop } from '../../utils/get-margin';

const StyledPagination = styled.div`
  display: flex;
  ${marginTop(2)}
`;

const Container = styled.div`
  width: 50%;
`;

const Previous = styled(Container)`
  text-align: left;
`;

const Next = styled(Container)`
  text-align: right;
`;

const StyledLink = styled(Link)`
  color: var(--color-secondary);
  font-size: 26px;
  font-weight: bold;
  &:hover,
  &:focus {
    color: var(--color-primary);
  }
  ${(props) => props.disabled && css`
  pointer-events: none;
  color: var(--color-gray700);
  `}
`;

type Props = {
  prevPagePath: string,
  nextPagePath: string,
  hasNextPage: boolean,
  hasPrevPage: boolean
};

const Pagination = ({
  prevPagePath,
  nextPagePath,
  hasNextPage,
  hasPrevPage
}: Props) => (
  <StyledPagination>
    <Previous>
      <StyledLink rel="prev" to={hasPrevPage ? prevPagePath : '/'} disabled={!hasPrevPage}>{PAGINATION.PREV_PAGE}</StyledLink>
    </Previous>
    <Next>
      <StyledLink rel="next" to={hasNextPage ? nextPagePath : '/'} disabled={!hasNextPage}>{PAGINATION.NEXT_PAGE}</StyledLink>
    </Next>
  </StyledPagination>
);

export default Pagination;
