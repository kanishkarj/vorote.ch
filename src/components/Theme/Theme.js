import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  fonts: {
    body: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    heading: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    monospace: 'Menlo, monospace',
  },
  fontWeights: {
    body: 400,
    heading: 600,
    bold: 700,
  },
  fontSizes: {
    base: 16,
    small: 14,
  },
  lineHeight: 1.625,
  spacing: Math.round(16 * 1.625),
  buttons: {
    height: 35,
    borderRadius: 20,
  },
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  }
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

export default Theme;
