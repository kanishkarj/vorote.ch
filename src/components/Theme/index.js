// @flow strict
export { default } from './Theme';
