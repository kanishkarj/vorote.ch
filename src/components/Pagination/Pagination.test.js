// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './Pagination';
import 'jest-styled-components'
import Theme from '../Theme';

describe('Pagination', () => {
  const props = {
    prevPagePath: '/page/1',
    nextPagePath: '/page/3',
    hasNextPage: true,
    hasPrevPage: true
  };

  it('renders correctly', () => {
    const tree = renderer.create(
    <Theme>
      <Pagination {...props} />
    </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
