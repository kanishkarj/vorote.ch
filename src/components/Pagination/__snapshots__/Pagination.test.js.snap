// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 52px;
}

.c1 {
  width: 50%;
  text-align: left;
}

.c3 {
  width: 50%;
  text-align: right;
}

.c2 {
  color: var(--color-secondary);
  font-size: 26px;
  font-weight: bold;
}

.c2:hover,
.c2:focus {
  color: var(--color-primary);
}

<div
  className="c0"
>
  <div
    className="sc-AxirZ c1"
  >
    <a
      className="c2"
      disabled={false}
      href="/page/1"
      rel="prev"
    >
      ← PREV
    </a>
  </div>
  <div
    className="sc-AxirZ c3"
  >
    <a
      className="c2"
      disabled={false}
      href="/page/3"
      rel="next"
    >
      → NEXT
    </a>
  </div>
</div>
`;
